### PR TITLE
Run gopy bind within same process in go test

### PR DESCRIPTION
**Pull request by rameshg87**
This allows getting the real coverage in the code.

### CODE REVIEW

**Comment by rameshg87** [comment]
Coverage data: https://gist.github.com/rameshg87/1e117331c7deb5795c7798e3b0d0ab11

File: main.go [hunk 1]
```diff
@@ -31,19 +31,26 @@ func init() {
 	}
 }
 
-func main() {
-	err := app.Flag.Parse(os.Args[1:])
+func runArgs(argsStr []string) error {
```
Comment (sbinet) [inline main.go]: I would call this function just `run` and its argument `args`.

File: main.go [hunk 2]
```diff
@@ -31,19 +31,26 @@ func init() {
...
+	err := app.Flag.Parse(argsStr)
 	if err != nil {
 		log.Printf("error parsing flags: %v\n", err)
-		os.Exit(1)
+		return err
```
Comment (sbinet) [inline main.go]: as `run` returns an `error` now, I would actually remove the `log.Printf` above and return the "decorated" error like so:

```go
return fmt.Errorf("could not parse flags: %v", err)
```

File: main.go [hunk 3]
```diff
@@ -31,19 +31,26 @@ func init() {
...
 	}
 
 	args := app.Flag.Args()
 	err = app.Dispatch(args)
 	if err != nil {
 		log.Printf("error dispatching command: %v\n", err)
-		os.Exit(1)
+		return err
```
Comment (sbinet) [inline main.go]: same comment as above.

File: main.go [hunk 4]
```diff
@@ -31,19 +31,26 @@ func init() {
...
 	}
+	return nil
+}
 
+func main() {
+	err := runArgs(os.Args[1:])
+	if err != nil {
+		os.Exit(1)
```
Comment (sbinet) [inline main.go]: and now, I would call:

```go
log.Print(err)
os.Exit(1)
```

or, directly:
```go
log.Fatal(err)
```

File: main.go (hunk 1 above)
Comment (rameshg87) [reply main.go]: Done. I had to rename the args variable in the method to appArgs to avoid conflict in naming.

File: main.go (hunk 2 above)
Comment (rameshg87) [reply main.go]: Done.

File: main.go (hunk 3 above)
Comment (rameshg87) [reply main.go]: Done.

File: main.go (hunk 4 above)
Comment (rameshg87) [reply main.go]: Done.

**Comment by sbinet** [comment]
still LGTM.

could you squash the two commits into one that has the following header message?

```
gopy: refactor main to assess coverage

blabla...
```

then, please send another PR against [go-python/license](https://github.com/go-python/license) adding yourself as an author+contributor.

(and I'll merge these 2 PRs)

thanks!

**Comment by rameshg87** [comment]
Can you please review and merge this pull request if it looks fine?

**Comment by sbinet** [comment]
@rameshg87 as I said in https://github.com/go-python/gopy/pull/153#issuecomment-355768247, this PR is good to go.
I'll merge it in once you've sent another PR against [go-python/license](https://github.com/go-python/license) adding yourself as an author+contributor.
this is a needed "bureaucracy" one-time annoyance so all the code under go-python is correctly licensed.

thanks :)

**Comment by sbinet** [comment]
needs go-python/license#8.

**Comment by sbinet** [comment]
thanks!

**Pull request merged**